### PR TITLE
Move log file writing to a separate thread

### PR DESCRIPTION
**Pull request by jherico**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 63dfd570f198e243709bad1a54f0af1821fe3a49 available at: https://gist.github.com/c2ceeb8be51371850c5f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5285.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5285.dmg

File: interface/src/FileLogger.cpp [hunk 1]
```diff
@@ -33,16 +57,14 @@ FileLogger::FileLogger(QObject* parent) :
...
 
-    QFile file(_fileName);
-    if (file.open(QIODevice::WriteOnly | QIODevice::Append | QIODevice::Text)) {
-        QTextStream out(&file);
-        out <<  message;
-    }
+void FileLogger::addMessage(const QString& message) {
+    _persistThreadInstance->queueItem(message);
+    emit logReceived(message);
+    //_logData += message;
```
Comment (ZappoMan) [inline interface/src/FileLogger.cpp]: is this dead code? what was the _logdata variable doing before?

File: interface/src/FileLogger.h [hunk 2]
```diff
@@ -13,23 +13,25 @@
...
 class FileLogger : public AbstractLoggerInterface {
     Q_OBJECT
 
 public:
     FileLogger(QObject* parent = NULL);
+    virtual ~FileLogger();
 
-    virtual void addMessage(QString);
+    virtual void addMessage(const QString&);
     virtual QString getLogData() { return _logData; }
```
Comment (ZappoMan) [inline interface/src/FileLogger.h]: looks like this is dead code? are their any callers of getLogData()? The _logData variable isn't being set/added to.

Comment (ZappoMan) [inline interface/src/FileLogger.h]: this appears to be used by the log dialog box. I think this is now broken. People will definitely notice that and complain.

File: interface/src/FileLogger.cpp (hunk 1 above)
Comment (ZappoMan) [inline interface/src/FileLogger.cpp]: After testing, I can confirm that not having this line of code causes the log dialog box to not have the full log when you open it. We need to fix this bug before we merge.

File: interface/src/FileLogger.h (hunk 2 above)
Comment (jherico) [inline interface/src/FileLogger.h]: fixed.

File: interface/src/FileLogger.cpp (hunk 1 above)
Comment (jherico) [inline interface/src/FileLogger.cpp]: fixed

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2fff560ff9dadfcbb2d8b60eebd881676812cbae available at: https://gist.github.com/c2ceeb8be51371850c5f<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5285.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5285.dmg

**Comment by ZappoMan** [comment]
tested on windows and mac

**Pull request merged**